ui-components/src/blocks/ImageCard.tsx

```diff
@@ -64,48 +84,125 @@ const ImageCard = (props: ImageCardProps) => {
+                      symbol={tag.iconType}
+                      fill={tag.iconColor ?? IconFillColors.primary}
+                    />
+                  )}
+                  {tag.text}
+                </Tag>
+              </React.Fragment>
             )
-          }
-
-          return <React.Fragment key={index}>{tagContent}</React.Fragment>
-        })}
+          })}
+        </div>
+        <figure className={innerClasses.join(" ")}>
+          {props.imageUrl ? (
+            <img
+              src={props.imageUrl}
+              alt={props.description || t("listings.buildingImageAltText")}
+            />
+          ) : props.images && displayedImages ? (
+            displayedImages.map((image, index) => (
+              <img
+                key={index}
+                src={image.thumbnailUrl || image.mobileUrl || image.url}
+                alt={
+                  image.description
+                    ? image.description
+                    : `${props.description || ""} - photo ${index + 1}`
+                }
+              />
+            ))
+          ) : (
+            <div className={"image-card__placeholder"} />
+          )}
+          {props.images && props.images.length > 1 && (
+            <>
+              {props.images && props.images.length > 3 && (
+                <div className="image-card__more-images">
+                  <Icon symbol="plus" size="xlarge" />
+                  {props.moreImagesLabel && (
+                    <span>
+                      {props.images.length - 2} {props.moreImagesLabel}
+                    </span>
+                  )}
+                </div>
+              )}
+              <button
+                aria-label={
+                  props.moreImagesDescription &&
+                  `${props.images.length - 2} ${props.moreImagesDescription}`
+                }
+                data-test-id="open-modal-button"
+                onClick={() => {
+                  setShowModal(true)
+                }}
+              ></button>
+            </>
+          )}
+        </figure>
+        {getStatuses()}
       </div>
-      <figure className="image-card">
-        {props.imageUrl ? (
-          <img src={props.imageUrl} alt={props.description || ""} />
-        ) : (
-          <div className={"image-card__placeholder"} />
-        )}
-      </figure>
-      {getStatuses()}
-    </div>
+      {props.images && props.images.length > 1 && (
+        <Modal
+          open={showModal}
+          title={props.modalAriaTitle || "Images"}
+          scrollable={true}
+          onClose={() => setShowModal(!showModal)}
+          className="image-card__overlay"
+          modalClassNames="image-card__gallery-modal"
+          headerClassNames="sr-only"
+          actions={[
+            <Button onClick={() => setShowModal(!showModal)}>
+              {props.modalCloseLabel || "Close"}
+            </Button>,
+          ]}
+        >
+          {props.images &&
```

Comment:
We shouldn't need this additional check for props.images, since it's done above.